pkg/controller/cluster/cluster_controller.go

```diff
@@ -168,3 +220,83 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (reconcile.Resul
 	}
 	return reconcile.Result{}, nil
 }
+
+// listChildren returns a list of Deployments, Sets, and Machines than have an ownerref to the given cluster
+func (r *ReconcileCluster) listChildren(ctx context.Context, cluster *clusterv1.Cluster) ([]runtime.Object, error) {
+	var children []runtime.Object
+
+	ns := cluster.GetNamespace()
+	opts := metav1.ListOptions{
+		LabelSelector: labels.FormatLabels(
+			map[string]string{clusterv1.MachineClusterLabelName: cluster.GetName()},
+		),
+	}
+
+	dfunc := func(_ context.Context, m metav1.ListOptions) (runtime.Object, error) {
+		return r.clusterClient.MachineDeployments(ns).List(m)
+	}
+	sfunc := func(_ context.Context, m metav1.ListOptions) (runtime.Object, error) {
+		return r.clusterClient.MachineSets(ns).List(m)
+	}
+	mfunc := func(_ context.Context, m metav1.ListOptions) (runtime.Object, error) {
+		return r.clusterClient.Machines(ns).List(m)
+	}
+
+	deployments, err := pager.New(dfunc).List(ctx, opts)
```

Comment:
Thinking more about using `r.clusterClient`... if we use `r.Client.List` instead, it will take advantage of the shared informer's cache for each type. That's probably better. WDYT?

Reply:
[per Slack discussion](https://kubernetes.slack.com/archives/C8TSNPY4T/p1563904482131800?thread_ts=1563904056.129800&cid=C8TSNPY4T), we've decided to keep the clusterClient for easier use of the pager